File: app/src/main/java/com/imceits/aungtuntun/alephcodeassignment/ui/FriendViewModel.java
package com.imceits.aungtuntun.alephcodeassignment.ui;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.imceits.aungtuntun.alephcodeassignment.data.ToolRepository;
import com.imceits.aungtuntun.alephcodeassignment.data.entity.Friends;

import java.util.List;

import javax.inject.Inject;

public class FriendViewModel extends ViewModel {

    private final ToolRepository toolRepository;
    private LiveData<List<Friends>> friendsList;

    @Inject
    FriendViewModel(ToolRepository toolRepository) {
        this.toolRepository = toolRepository;
        friendsList = this.toolRepository.getAllFriends();
    }

    public LiveData<List<Friends>> getFriendsList() {
        return friendsList;
    }
}

File: app/src/main/java/com/imceits/aungtuntun/toolsbox/ui/FriendViewModel.kt
package com.imceits.aungtuntun.toolsbox.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.imceits.aungtuntun.toolsbox.data.ToolDatabase
import com.imceits.aungtuntun.toolsbox.data.ToolRepository
import com.imceits.aungtuntun.toolsbox.data.entity.Friends

class FriendViewModel(application: Application) : AndroidViewModel(application) {

    val friendsList: LiveData<List<Friends>>
    private val repository: ToolRepository

    init {
        val toolsRentDao = ToolDatabase.getInstance(application, viewModelScope).toolsRentDao()
        repository = ToolRepository( toolsRentDao)
        friendsList = repository.friendList
    }

}
